### ISSUE TITLE
react/no-this-in-sfc: `this` inside a nested arrow callback prop (e.g. `ref` handler) of an SFC is not flagged

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25615]
## Summary

The native `react/no-this-in-sfc` rule flags a direct `this` member expression in a JSX spread attribute of an arrow-function SFC declared inside a class constructor, but misses `this` used inside a nested arrow-function callback passed as a JSX attribute of that same SFC (a `ref` handler). `eslint-plugin-react` 7.37.5 flags both usages, under both ESLint 7 and ESLint 9. As a consequence, an `eslint-disable-next-line react/no-this-in-sfc` directive over the nested-callback line — required for ESLint parity — is reported by `--report-unused-disable-directives` as unused.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`repro.tsx`:

```tsx
import React from 'react'

class ItemAdapter {
	el: unknown
	itemRef: unknown = null

	constructor() {
		// ElementWrapper is an arrow-function SFC assigned to a local const,
		// then stored on the instance. It closes over `this`.
		const ElementWrapper = (props: {label: string}) => (
			<div
				ref={ref => {
					this.itemRef = ref // <-- should be flagged: `this` in an SFC
				}}
				{...this.getBasicProps()} // <-- should be flagged: `this` in an SFC
			>
				{props.label}
			</div>
		)

		this.el = ElementWrapper
	}

	getBasicProps() {
		return {}
	}
}
```

`.oxlintrc.json`:

```json
{
	"plugins": ["react"],
	"categories": {"correctness": "off"},
	"rules": {"react/no-this-in-sfc": "error"}
}
```

Run:

```
oxlint -c .oxlintrc.json --disable-nested-config repro.tsx
```

## Expected

Two `react/no-this-in-sfc` diagnostics — one at `this.itemRef = ref` (13:6) and one at `...this.getBasicProps()` (15:9). ESLint 7.23.0 (legacy config) and ESLint 9.39.4 (flat config), both with `eslint-plugin-react` 7.37.5, report exactly those two errors ("Stateless functional components should not use `this`").

## Actual

```
repro.tsx:15:9: error react(no-this-in-sfc): Stateless functional components should not use `this`
```

Only the JSX spread usage is flagged (anchor matches ESLint exactly). The `this.itemRef = ref` usage at line 13, inside the arrow callback passed to the `ref` prop, is not reported — the SFC-body traversal appears not to descend into nested function/arrow expressions used as JSX attribute values. With `// eslint-disable-next-line react/no-this-in-sfc` above both usages and `--report-unused-disable-directives`, the directive covering the nested-callback line is reported as "Unused eslint-disable directive (no problems were reported)" even though it is required under ESLint.

## Cross-checked

| Engine | Result on `repro.tsx` |
| --- | --- |
| ESLint 7.23.0 + eslint-plugin-react 7.37.5 (@typescript-eslint/parser 6.21.0) | 2 errors: 13:6 and 15:9 |
| ESLint 9.39.4 (flat config) + eslint-plugin-react 7.37.5 (typescript-eslint 8.62.1) | 2 errors: 13:6 and 15:9 |
| oxlint 1.78.0 (native rule) | 1 error: 15:9 only; line 13 missed |
